#[187364740] A seller should be able to delete a specific item from their collection

Yvan-David opened the pull request:
#### What does this PR do?

 - endpoint handles scenario 1 where a user token is of role seller but the item does not exist
 - endpoint handles scenario 2 where a user token is of role seller and the item already exist and the body is validated
 - endpoint handles scenario 3 when request body fails any validation checks
 - validate all inputs for sanity

#### Description of Task to be completed?
     As a Seller
    I should be able to delete a specific item in my colletion
    Such that I can remove items I no longer deal in/were erroneously added

#### How should this be manually tested?

  -  Using swagger documentation
  -  Using postman
  - Link to Loom recording: [https://www.loom.com/share/6aec148a6e6a4ee2b1a6b5d846d52f83?sid=9abbdd88-148e-4542-9132-bb1fc19af22c](https://www.loom.com/share/6aec148a6e6a4ee2b1a6b5d846d52f83?sid=9abbdd88-148e-4542-9132-bb1fc19af22c)
#### Any background context you want to provide?

#### What are the relevant pivotal tracker/Trello stories?

   -  #187364740

#### Screenshots (if appropriate)

#### Questions:

Bananayosostene commented:
There are same checks which are not successful it caused by coverage, so increase coverage

Yvan-David commented:
> There are same checks which are not successful it caused by coverage, so increase coverage

they are unsuccessful because coverage have decreased and it can not always be 100

AngeloChristian1 commented on src/controllers/productController.ts:
```diff
@@ -0,0 +1,33 @@
+     * @param {Response} res - Express response object
+     */
+
+    static async deleteProduct(req: Request, res: Response) {
```
I think you nedd to first check if the product is in their collection

EricNiyo73 commented on src/database/models/productModel.ts:
```diff
@@ -0,0 +1,96 @@
+  images: string[]
+  sellerId: number
+}
+
```
Why this model does not have a collectionID, The seller can create or delete a specific item from their collection, which means foreign key from Collection Model is needed.

EricNiyo73 commented on src/controllers/productController.ts:
```diff
@@ -0,0 +1,33 @@
+        if(product){
+          const result: number = await deleteProductById(id, productId)
+          if (result !== 0) {
+            return res.status(200).json({message: "item deleted", productDeleted: product})
+          }
```
How the deleted Product can be displayed??
May be you can return only message instead

Furaha-Justine committed:
feat(seller deletes specific item): implement endpoints for delete specific item

 - endpoint handles scenario 1 where a user token is of role seller but the item does not exist
 - endpoint handles scenario 2 where a user token is of role seller and the item already exist and the body is validated
 - endpoint handles scenario 3 when request body fails any validation checks
 - validate all inputs for sanity

[Delivers #187364740]

AngeloChristian1 commented on src/controllers/productController.ts:
```diff
@@ -118,4 +119,28 @@ export default class productController {
+      const productId = req.params.productId;
+      const sellerId = req.user?.id
+      const product = await getProductById(sellerId, productId)
+      if(product){
+        const result: number = await deleteProductById(sellerId, productId)
+        if (result !== 0) {
+          return res.status(200).json({message: "item deleted"})
+        }
+      } else {
+        return res.status(410).json({message: "item not found"});
+      }
```
Won't this be a better implementation?
```suggestion
      if(!product){
      return res.status(410).json({message: "item not found"});
      }
      await deleteProductById(sellerId, productId)
       return res.status(200).json({message: "item deleted"})
```

Yvan-David replied:
thanks for the suggestion

AngeloChristian1 reviewed (commented):
Great work, please see if you can work on the suggestions
Regards

munezerobagira commented on src/controllers/productController.ts:
```diff
@@ -7,6 +7,7 @@ import {
 } from '../config/index'
 import Product from '../database/models/productModel'
 import Collection from '../database/models/collectionModel'
+import { deleteProductById, getProductById } from '../services/productServices'
```
```suggestion
import { deleteProductById, getProductById } from '../services/productServices'
```
is productServices or productService

munezerobagira commented on src/docs/productDocs.ts:
```diff
@@ -84,4 +84,44 @@
  *                   example: Collection not found
  *       '500':
  *         description: Internal server error
+ * /collections/{productId}:
```
```suggestion
 * /collections/products/{productId}:
```

munezerobagira commented on src/controllers/productController.ts:
```diff
@@ -118,4 +119,28 @@ export default class productController {
+          return res.status(200).json({message: "item deleted"})
+        }
+      } else {
+        return res.status(410).json({message: "item not found"});
```
```suggestion
        return res.status(404).json({message: "item not found"});
```
 check  if this is correct  status code
